### PR TITLE
rustc: switch to rust edition 2021

### PR DESCRIPTION
**Pull request by q9f**
from 2018

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -3539,11 +3539,11 @@ dependencies = [
 
 [[package]]
 name = "jsonwebtoken"
-version = "8.0.1"
+version = "7.2.0"
```
Comment (q9f) [inline Cargo.lock:3542]: I will never understand what is causing this.

**Comment by lemmih** [comment]
CI is failing due to transient network errors.

**Review by LesnyRumcajs — COMMENTED** [review]
Do you know if it's possible to define this only in the workspace root and let it be propagated without repeating the same setting everywhere?

**Comment by q9f** [comment]
I don't think it's possible because it will always use `2015` as the default unless indicated otherwise.

**Comment by LesnyRumcajs** [comment]
> I don't think it's possible because it will always use `2015` as the default unless indicated otherwise.

Indeed, seems this is still not implemented. https://github.com/rust-lang/cargo/issues/5784

Hopefully one day we'll get rid of this boilerplate!

**Review by LesnyRumcajs — APPROVED** [review]

**Review by lemmih — APPROVED** [review]

**Pull request merged**